### PR TITLE
[FLINK-30358][rest] Exception location contains TM resource id and link to TM pages.

### PR DESCRIPTION
**Pull request by JunRuiLee**
## What is the purpose of the change

At present, the web UI exception history page only displays the TM host and port. However, we generally need to search for problems according to the pod name or container ID.

Therefore, it is more convenient to add resource id (pod name on k8s, container id on yarn) to the location column and a link to the task manager id to jump to the task manager page.

## Brief change log

  - Exception location contains tm resource id.
  - Exception location links to taskManagers UI pages.

## Verifying this change

This change is already covered by existing tests, such as JobExceptionsHandlerTest.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

**Comment by zhuzhurk** [comment]
Thanks for making this change. The change looks good to me.
However, I would like to invite a web frontend expert to double confirm the changes to the web part.
@yangjunhan would you help to take a look at the changes?

File: flink-runtime-web/web-dashboard/src/app/pages/job/exceptions/job-exceptions.component.ts [hunk 1]
```diff
@@ -151,4 +153,10 @@ export class JobExceptionsComponent implements OnInit, OnDestroy {
         });
       });
   }
+
+  public navigateTo(taskManagerId: string): void {
```
[the remark this answers is not in the document]
Comment (JunRuiLee) [reply flink-runtime-web/web-dashboard/src/app/pages/job/exceptions/job-exceptions.component.ts]: updated!

File: flink-runtime-web/web-dashboard/src/app/pages/job/exceptions/job-exceptions.component.ts [hunk 2]
```diff
@@ -151,4 +153,10 @@ export class JobExceptionsComponent implements OnInit, OnDestroy {
...
+    if (taskManagerId != null) {
```
[the remark this answers is not in the document]
Comment (JunRuiLee) [reply flink-runtime-web/web-dashboard/src/app/pages/job/exceptions/job-exceptions.component.ts]: updated!

**Comment by JunRuiLee** [comment]
Hi @yangjunhan, I have updated this pr, PTAL. Thanks a lots.

**Review by zhuzhurk — APPROVED** [review]
Thanks for addressing the comments! @JunRuiLee 
And thanks for helping with the review. @yangjunhan 
LGTM. Merging.

**Pull request merged**